pkg/processor/trigger/kafka/trigger.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2018 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafka
+
+import (
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/errors"
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"github.com/nuclio/nuclio/pkg/processor/trigger"
+	"github.com/nuclio/nuclio/pkg/processor/worker"
+
+	"github.com/bsm/sarama-cluster"
+	"github.com/nuclio/logger"
+)
+
+type kafka struct {
+	trigger.AbstractTrigger
+	configuration  *Configuration
+	kafkaConfig    *cluster.Config
+	consumer       *cluster.Consumer
+	shutdownSignal chan struct{}
+}
+
+func newTrigger(parentLogger logger.Logger,
+	workerAllocator worker.Allocator,
+	configuration *Configuration) (trigger.Trigger, error) {
+	var err error
+
+	newTrigger := &kafka{
+		configuration: configuration,
+	}
+
+	newTrigger.AbstractTrigger = trigger.AbstractTrigger{
+		ID:              configuration.ID,
+		Logger:          parentLogger.GetChild(configuration.ID),
+		WorkerAllocator: workerAllocator,
+		Class:           "async",
+		Kind:            "kafka-cluster",
+	}
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create abstract stream")
+	}
+
+	newTrigger.Logger.DebugWith("Creating consumer", "url", configuration.URL)
+
+	newTrigger.kafkaConfig, err = newTrigger.newKafkaConfig(configuration.Attributes)
```

Comment:
You should not use `Attributes` in the trigger. The factory's responsibility is to convert the unstructured configuration into a structured one (and you do this)